code/datums/components/nanites.dm

```diff
@@ -246,6 +248,10 @@
 /datum/component/nanites/proc/get_programs(datum/source, list/nanite_programs)
 	nanite_programs |= programs
 
+/datum/component/nanites/proc/on_examine(datum/source, mob/user)
+	if(!stealth && visible_examine)
+		to_chat(user, "<span class='notice'>Their eyes shine with an electric fluid.</span>")
```

Comment:
This line will be outputted regardless of the user's eyes being covered and/or missing.
Add early returns if `user.is_eyes_covered()` or `(iscarbon(user) && !user.getorganslot(ORGAN_SLOT_EYES))` are TRUE.

Reply:
Oh, another thing: since carbon & co examine strings are batched, just return the text without `to_chat()` once you add `msg += SEND_SIGNAL(src, COMSIG_PARENT_EXAMINE, user)` to said procs.